ecommerce/api.py

```diff
@@ -164,6 +167,21 @@ def fulfill_completed_order(
     )
 
 
+def add_fulfilled_order_to_paid_courserun(order):
+    """
+    When order is fulfilled or reviewed, record it to paid courserun table if not exist
+    """
+    order = Order.objects.get(id=order.id)
+    if order.STATE.FULFILLED or order.STATE.REVIEW:
+        for run in order.purchased_runs:
+            if not PaidCourseRun.objects.filter(
+                order=order, course_run=run, user=order.purchaser
+            ).exists():
+                PaidCourseRun.objects.create(
+                    order=order, course_run=run, user=order.purchaser
+                )
```

Comment:
Small change: there's a `update_or_create()` [function](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#update-or-create) that could be used here. (Or, for that matter, `get_or_create()` if you don't want it to update the timestamps.)